`make start-gardenlet` fails for a new Seed

**How to categorize this issue?**
<!--
Please select area, kind, and priority for this issue. This helps the community categorizing it.
Replace below TODOs or exchange the existing identifiers with those that fit best in your opinion.
If multiple identifiers make sense you can also state the commands multiple times, e.g.
  /area control-plane
  /area auto-scaling
  ...

"/area" identifiers:     audit-logging|auto-scaling|backup|certification|control-plane-migration|control-plane|cost|delivery|dev-productivity|disaster-recovery|documentation|high-availability|logging|metering|monitoring|networking|open-source|ops-productivity|os|performance|quality|robustness|scalability|security|storage|testing|usability|user-management
"/kind" identifiers:     api-change|bug|cleanup|discussion|enhancement|epic|impediment|poc|post-mortem|question|regression|task|technical-debt|test
-->
/area dev-productivity
/kind bug
/kind regression

**What happened**:
I created a new Seed in my local setup. `make start-gardenlet` failed for it with:
```
% make start-gardenlet SEED_NAME=foo

priorityclass.scheduling.k8s.io/gardener-system-critical created
error: failed to create token: serviceaccounts "gardenlet" not found
make: *** [start-gardenlet] Error 1
```

The script tries to find the garden/gardenlet ServiceAccount:
https://github.com/gardener/gardener/blob/6d18b043a3852034e571ccc9f5e265db1923fd45/hack/local-development/start-gardenlet#L172

https://github.com/gardener/gardener/pull/7108 modified the ServiceAccount, Role and RoleBinding resource to do no longer hard-code the namespace but to use `{{ .Release.Namespace }}`. 
Current the `helm template` invocations in the start-gardenlet script do not pass a namespace, hence all gardenlet resources are created in the default namespace:
```yaml
---
# Source: gardenlet/templates/serviceaccount.yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: gardenlet
  namespace: default
  labels:
    app: gardener
    role: gardenlet
    chart: "gardenlet-0.1.0"
    release: "gardenlet"
    heritage: "Helm"
```

**What you expected to happen**:
`make start-gardenlet` to work for a new Seed as before.

**How to reproduce it (as minimally and precisely as possible)**:
See above.

**Anything else we need to know?**:

**Environment**:

- Gardener version: 6d18b043a3852034e571ccc9f5e265db1923fd45
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Others:

--- a/hack/local-development/start-gardenlet
+++ b/hack/local-development/start-gardenlet
@@ -152,14 +152,14 @@ kubectl   --kubeconfig="$SEED_KUBECONFIG" get    namespace garden &>/dev/null ||
   kubectl --kubeconfig="$SEED_KUBECONFIG" create namespace garden
 
 # apply RBAC resources in seed cluster
-helm template gardenlet "$GARDENLET_CHARTS_DIR" -f "$gardenletChartValues" |
+helm template gardenlet "$GARDENLET_CHARTS_DIR" -f "$gardenletChartValues" -n garden |
   yq eval 'select(.apiVersion=="rbac.authorization.k8s.io/v1")' - |
   kubectl --kubeconfig="$SEED_KUBECONFIG" auth reconcile --remove-extra-permissions --remove-extra-subjects -f -
-helm template gardenlet "$GARDENLET_CHARTS_DIR" -s templates/serviceaccount.yaml -f "$gardenletChartValues" |
+helm template gardenlet "$GARDENLET_CHARTS_DIR" -s templates/serviceaccount.yaml -f "$gardenletChartValues" -n garden |
   kubectl --kubeconfig="$SEED_KUBECONFIG" apply -f -
 
 # apply gardener-system-critical priority class in seed cluster
-helm template gardenlet "$GARDENLET_CHARTS_DIR" -s templates/priorityclass.yaml -f "$gardenletChartValues" |
+helm template gardenlet "$GARDENLET_CHARTS_DIR" -s templates/priorityclass.yaml -f "$gardenletChartValues" -n garden |
   kubectl --kubeconfig="$SEED_KUBECONFIG" apply -f -
 
 rm -f "$gardenletChartValues"

